# Q1

Ideas that can be tested must be supported by data and ideas that cannot be tested are generally people's subjective ideas that cannot be supported by data. A good null hypothesis is that something can be tested and there is a lot of data to support the original hypothesis, which has no effect or no significant difference between groups The alternative hypothesis is to assume something happened, and because they are opposites, we can only accept one.

# Q2


When we do a statistical test, we collect data from a small group which is a sample to make guesses about a larger group(population). For example, if we want to collect the data of average mark of English class, we will collect the whole lecture parts instead of one small tutorial. The numbers we calculate from the sample are called sample statistics, but we are care about is the true value for the whole population which is the population parameter. This can help us understand the whole group not only the sample.


Chatgpt link: https://chatgpt.com/share/670f2703-bf14-8009-b73f-813035d1cfc2

Summary:

In hypothesis testing, we use data from a sample (a small group) to make conclusions about a population (the whole group). The results of a test are focused on what we can learn about the population parameter (the true value for the whole population) based on the sample statistic (the number calculated from the sample). The key distinction is that we perform these tests to understand the population, even though we only use a small part of it (the sample) to draw conclusions. The purpose of hypothesis testing is to evaluate whether there is enough evidence to reject the null hypothesis (which usually states there is no effect or difference) in favor of the alternative hypothesis (which suggests there is an effect or difference). The conclusions are always about the population, not just the sample.

# Q3

When calculating the p-value, we begin by assuming that the null hypothesis(Ho) is true, meaning there is no effect or difference between the groups. So we then check if the p-value is extremely small, which would indicate that the null hypothesis is likely wrong. In that case, we reject H₀ and consider the alternative hypothesis(Ha). It's difficult to directly prove that Ha is correct so we test H0 to see if there is evidence strongly rejecting it. The two hypotheses are opposites of each other.

# Q4

A small p-value shows that the evidence from our data is very ridiculous so we start to doubt the null hypothesis. For example, from the video, if the dog is innocent in this case, the garbage cover on his head is done by someone else but we know that the probability of the neighborhood kid coming to our house and putting a garbage bag over his head is very small so it can only be done by the dog itself so we reject H0 and p value is very small, which is very ridiculous.

In [12]:
# Q5

import numpy as np


n_simulations = 10000
n_trials = 124
observed_right_tilts = 80

p_null = 0.5

simulations = np.random.binomial(n=n_trials, p=p_null, size=n_simulations)

p_value = np.mean(simulations >= observed_right_tilts)

print(f"Simulated p-value: {p_value}")


Simulated p-value: 0.0007


P-value is smaller than 0.001, therefore there is very strong evidence against the null hypothesis, which is reject H0.

# Q6

It can't prove that is false because we should use the p-value table which is decision rule to see each intervals instead of a point. Even if our evidence is very strong, the answer is not 100% correct. Because there's also error for type1 and type2. Type1 error means when we reject the null hypothesis H0, when it is actually true. Type2 error means we fail to reject the null hypothesis when the alternative hypothesis Ha is actually true. Therefore, although smaller p-value could has less probability to make mistakes but it is still wrong.

In [13]:
# Q7


import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

np.random.seed(1500)

data = {
    "PatientID": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    "InitialHealthScore": [84, 78, 83, 81, 81, 80, 79, 85, 76, 83],
    "FinalHealthScore": [86, 86, 80, 86, 84, 86, 86, 82, 83, 84]
}

df = pd.DataFrame(data)

# Calculate change in health scores
df['HealthScoreChange'] = df['FinalHealthScore'] - df['InitialHealthScore']

# Bootstrap sampling
n_iterations = 10000
boot_means = []
observed_mean = df['HealthScoreChange'].mean()

for i in range(n_iterations):
    boot_sample = df['HealthScoreChange'].sample(frac=1, replace=True)
    boot_means.append(boot_sample.mean())

# Confidence interval for two-tailed test
confidence_interval = np.percentile(boot_means, [2.5, 97.5])

# For the one-tailed test (checking if the mean change is greater than 0)
# Calculate p-value for one-tailed test (greater than 0)
p_value_one_tailed = np.sum(np.array(boot_means) >= observed_mean) / n_iterations

print(f"Observed Mean: {observed_mean}")
print(f"95% Confidence Interval (Two-tailed): {confidence_interval}")
print(f"One-tailed p-value (Mean Health Score Change > 0): {p_value_one_tailed}")

Observed Mean: 3.3
95% Confidence Interval (Two-tailed): [0.9 5.5]
One-tailed p-value (Mean Health Score Change > 0): 0.5278


# Q7(Continued)


In the original code, a two-tailed p-value checks how extreme the observed mean is in both directions (positive and negative). For a one-tailed test, we only care about one side, specifically whether the mean is greater than 0. To do this, we compare the observed mean to the simulated (bootstrap) distribution and count how many of the simulated results are greater than or equal to the observed mean.

For the one-tailed test, the p-value is usually smaller because we are only looking at one side of the distribution (greater than or less than 0), while a two-tailed test splits the significance level between both sides, making it harder to detect a result in just one direction.

Chatgpt link: https://chatgpt.com/share/670f2703-bf14-8009-b73f-813035d1cfc2


Summary:

In the previous discussion, we examined how to modify a two-tailed hypothesis test to a one-tailed test using code. The key changes involved:

Shifting focus from checking for a general difference in both directions (two-tailed) to checking for a specific directional difference (one-tailed).
Adjusting the p-value calculation to reflect this focus, where we only consider values greater than or less than the observed mean, depending on the hypothesis.
In a one-tailed test, the interpretation changes because you're only testing one direction (e.g., whether the mean is greater than 0). This makes the test more sensitive and typically results in a smaller p-value compared to the two-tailed test, where the alpha is split between both tails of the distribution.

# Q8

The students are guessing, meaning they have a 50% chance of correctly identifying whether tea or milk was poured first. 
H0: p = 0.5, where p is the proportion of students who correctly identify the order.

The students are better than random guessing in identifying whether tea or milk was poured first.
Ha: p >= 0.5

The observed proportion of correct answers is p head = 49/80 = 0.6125.

In [14]:
import numpy as np

# Set seed for reproducibility
np.random.seed(42)

# Parameters
n_students = 80  # Total number of students
n_correct = 49  # Number of students who correctly identified the order
p_null = 0.5  # Null hypothesis proportion (50% guessing)
n_simulations = 10000  # Number of simulations

# Simulate 10,000 experiments under the null hypothesis (50% guessing)
simulated_results = np.random.binomial(n=n_students, p=p_null, size=n_simulations)

# Calculate p-value: proportion of simulations where number of correct answers >= 49
p_value = np.mean(simulated_results >= n_correct)

# Output p-value
print(f"Simulated p-value: {p_value}")


Simulated p-value: 0.0294


Since the p-value (0.0294) is less than the common significance level of 0.05 and greater than 0.01, we have moderate evidence to reject the null hypothesis. This suggests that the STA130 students may indeed be better than random guessing at identifying whether the milk or tea was poured first.

# Q9

Yes, I have reviewed the course wiki-textbook, which has been helpful in clarifying many concepts. I’ve also interacted with a ChatBot to address some of the material that I didn’t fully grasp at first.